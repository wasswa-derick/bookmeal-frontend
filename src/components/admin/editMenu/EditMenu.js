import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import Loader from "react-loader";
import { getMenu } from "../../../actions/menus";
import AddMenuForm from "../addMenu/addMenuForm";

/**
 * @export
 * @class EditMenuPage
 * @extends {React.Component}
 */
export class EditMenu extends React.Component {
  state = {
    loaded: false,
    errors: {}
  };
  componentDidMount = () => {
    const { id } = this.props.match.params;
    this.props
      .getMenu(id)
      .then(() => {
        this.setState({ loaded: true });
      })
      .catch(err => {
        console.log(err);
        // TODO: show errors
      });
  };
  onSubmit = () => {};
  onChange = () => {};
  handleDateChange = () => {};
  handleFileUpload = () => {};
  checked = () => {};

  /**
   * @returns {HTMLElement} div
   * @memberof EditMenu
   */
  render() {
    const { loaded, errors } = this.state;
    const { menu } = this.props;
    return (
      <div className="container">
        <Loader loaded={loaded}>
          <AddMenuForm
            title="Edit menu"
            menu={menu}
            errors={errors}
            onSubmit={this.onSubmit}
            onChange={this.onChange}
            handleDateChange={this.handleDateChange}
            handleFileUpload={this.handleFileUpload}
            checked={this.checked}
          />
        </Loader>
      </div>
    );
  }
}

EditMenu.propTypes = {
  getMenu: PropTypes.func.isRequired,
  menu: PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    menuDate: PropTypes.string.isRequired,
    url: PropTypes.string,
    meals: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number.isRequired,
        title: PropTypes.string.isRequired
      }).isRequired
    ).isRequired
  }).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired
    }).isRequired
  }).isRequired
};

const mapStateToProps = state => ({
  menu: state.menusReducer.menu
});

export default connect(mapStateToProps, { getMenu })(EditMenu);
