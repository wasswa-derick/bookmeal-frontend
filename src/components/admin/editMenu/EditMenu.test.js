import React from "react";
import { shallow } from "enzyme";
import { EditMenu } from "./EditMenu";

const err = {
  response: {
    status: 400,
    data: {
      errors: {}
    }
  }
};

const fn = () => Promise.resolve();

describe("EditMenu", () => {
  let wrapper;
  beforeEach(() => {
    const match = {
      params: { id: "1" }
    };
    const menu = {
      id: 0,
      title: "",
      description: "",
      meals: [],
      menuDate: ""
    };
    wrapper = shallow(<EditMenu match={match} menu={menu} getMenu={fn} />);
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
