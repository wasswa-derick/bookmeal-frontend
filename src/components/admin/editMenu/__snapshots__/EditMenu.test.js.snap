// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditMenu should render correctly 1`] = `
<div
  className="container"
>
  <Component
    component="div"
    loaded={false}
    loadedClassName="loadedContent"
    parentClassName="loader"
  >
    <AddMenuForm
      checked={[Function]}
      errors={Object {}}
      handleDateChange={[Function]}
      handleFileUpload={[Function]}
      menu={
        Object {
          "description": "",
          "id": 0,
          "meals": Array [],
          "menuDate": "",
          "title": "",
        }
      }
      onChange={[Function]}
      onSubmit={[Function]}
      title="Edit menu"
    />
  </Component>
</div>
`;
